src/arch/riscv/pagetable_walker.cc

```diff
@@ -490,8 +490,19 @@ Walker::WalkerState::recvPacket(PacketPtr pkt)
             Addr paddr = walker->tlb->translateWithTLB(vaddr, satp.asid, mode);
             req->setPaddr(paddr);
             walker->pma->check(req);
+
+            // If we are not in the m-mode, check pmp
+            // permissions
```

Comment:
You should *always* check permissions, IIRC. I think you can revoke permission from m-mode (through the locked bit?). If you do this, it's permanent until reboot, but it's still something that's in the spec, IIRC.

Reply:
I have updated the way we decide if we should check permissions. The update is based on a paragraph from the specs which mentions 3 conditions on which we should check permissions.